CMakeLists.txt

```
@@ -1,31 +1,20 @@
-cmake_minimum_required(VERSION 2.8.3)
-project(serial)
+cmake_minimum_required(VERSION 3.5.0)
 
-# Find catkin
-find_package(catkin REQUIRED)
+# General setup
+option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
+
+set(PROJ_SOVERSION 1)
+project(serial
+    VERSION 1.2.1
+    DESCRIPTION "Cross-platform, Serial Port library written in C++"
+    HOMEPAGE_URL "http://wjwwood.io/serial/"
+)
```

```suggestion
project(serial
    VERSION 1.2.1
    LANGUAGES CXX
    DESCRIPTION "Cross-platform, Serial Port library written in C++"
    HOMEPAGE_URL "http://wjwwood.io/serial/"
)
```
You can specify what compiled languages are being used to slightly speed up configuration. This means CMake won't bother checking for a C compile since no C compiler is needed.